portality/crosswalks/article_doaj_xml.py

```diff
@@ -0,0 +1,260 @@
+from portality.core import app
+from lxml import etree
+import re
+from portality.bll import exceptions
+from portality.crosswalks.exceptions import CrosswalkException
+from portality import models
+from datetime import datetime
+
+class DOAJXWalk(object):
+    format_name = "doaj"
+    schema_path = app.config.get("SCHEMAS", {}).get("doaj")
+
+    def __init__(self):
+        self.validation_log = ""
+
+        # load the schema into memory for more efficient usage in repeat calls to the crosswalk
+        if self.schema_path is None:
+            raise exceptions.IngestException(message="Unable to validate for DOAJXWalk, as schema path is not set in config")
+
+        try:
+            schema_file = open(self.schema_path)
+            schema_doc = etree.parse(schema_file)
+            self.schema = etree.XMLSchema(schema_doc)
+        except Exception as e:
+            raise exceptions.IngestException(message="There was an error attempting to load schema from " + self.schema_path, inner=e)
+
+    def validate_file(self, file_handle):
+        # first try to parse the file
+        try:
+            doc = etree.parse(file_handle)
+        except etree.XMLSyntaxError as e:   # although the treatment is the same, pulling this out so we remember what the primary kind of exception should be
+            raise CrosswalkException(message="Unable to parse XML file", inner=e)
+        except Exception as e:
+            raise CrosswalkException(message="Unable to parse XML file", inner=e)
+
+        # then pass the doc to the validator
+        valid = self.validate(doc)
+
+        if not valid:
+            msg = "Validation message from schema '{x}': {y}\n".format(x=DOAJXWalk.format_name, y=self.validation_log)
+            raise CrosswalkException(message="Unable to validate document with identified schema", inner_message=msg)
+
+        return doc
+
+    def validate(self, doc):
+        valid = self.schema.validate(doc)
+        if not valid:
+            el = self.schema.error_log.__repr__()
+            # strip the filename, as we don't want to leak the path to the UI
+            rx = "[\da-f]{32}.xml:(.*)"
```

Comment:
this is a regex - should it be in the regexes file?

Reply:
hm, I'm not sure.  I had thought the regex module would be for common re-usable regexes, like for DOIs, etc.  This regex is completely specific to this exact point in the code, so I don't think I would move it out.  What do you think?

Reply:
Yup, keep it here.